03/app.js

```diff
@@ -5,38 +5,62 @@ const root = createRoot(document.querySelector('#root'));
 
 class Article extends React.Component {
     state = {
+        comment: '',
         comments: [],
     }
-    
+
     render() {
-        const {title, body} = this.props;
+        const { title, body } = this.props;
+        const { comment } = this.state
         return (
             <article>
-                <h1>{ title }</h1>
-                <p>{ body }</p>
+                <h1>{title}</h1>
+                <p>{body}</p>
                 <section>
-                    <form>
+                    <form onSubmit={this.submitHandler}>
                         <div>
                             <label>
-                                <textarea 
-                                    style={{ "minWidth": "300px", "minHeight": "120px" }} 
-                                    name="content" 
+                                <textarea
+                                    onChange={this.changeHandler}
+                                    style={{ "minWidth": "300px", "minHeight": "120px" }}
+                                    name="content"
+                                    value={comment}
                                 />
                             </label>
                         </div>
                         <div><input type="submit" value="dodaj komentarz" /></div>
                     </form>
                     <ul>
-                        {/* tutaj komentarze jako <li/>, ps. tak wygląda komentarz do kodu w JSX */}
+                        {this.renderCommentsList()}
                     </ul>
                 </section>
             </article>
         )
     }
+
+    changeHandler = (e) => {
+        const { comment } = this.state  // trochę nie rozumiem dlaczego comment jest wyszarzony, jeśli zakomentuje tą linijkę, comment jest cały czas zdefiniowany- czy chodzi o closures? Jeśli tak, to wcześniej rozumiałam to tak, że mimo wszytsko zmienna w pierwszej kolejności pobierana jest z najbliższego scope, czyli z tej bieżącej linii kodu 42. 
```

Comment:
Przecież w tym wypadku ni korzystasz z tego elementu  - dlatego jest szary.
Zobacz linię 44: `this.setState({ comment: e.target.value })` - masz nowy obiekt, który ma właściwość o nazwie `comment` - to nie jest zmienna, która tworzysz przy destrukturyzacji.